cpp/src/arrow/device.h

```diff
@@ -165,6 +215,20 @@ class ARROW_EXPORT MemoryManager : public std::enable_shared_from_this<MemoryMan
   static Result<std::shared_ptr<Buffer>> ViewBuffer(
       const std::shared_ptr<Buffer>& source, const std::shared_ptr<MemoryManager>& to);
 
+  /// \brief Create a SyncEvent for exporting
```

Comment:
The references to exporting and importing in these docstrings seem a bit arbitrary, no? The goal here is to expose generally useful functionality.